front/app/modules/commercial/content_builder/admin/components/CraftComponents/Iframe/index.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+
+// components
+import { Box, IconTooltip, Input } from '@citizenlab/cl2-component-library';
+import Error from 'components/UI/Error';
+
+// intl
+import messages from '../../../messages';
+import { injectIntl } from 'utils/cl-intl';
+
+// craft
+import { useNode } from '@craftjs/core';
+
+// events
+import eventEmitter from 'utils/eventEmitter';
+
+interface Props {
+  url: string;
+  height: number;
+  hasError: boolean;
+  errorType?: string;
+  title?: string;
+}
+
+const Iframe = ({ url, height, hasError, title }: Props) => {
+  return (
+    <Box minHeight="26px">
+      {!hasError && url && (
+        <iframe src={url} title={title} width="100%" height={height} />
+      )}
+    </Box>
+  );
+};
+
+const IframeSettings = injectIntl(({ intl: { formatMessage } }) => {
+  const {
+    actions: { setProp },
+    url,
+    height,
+    id,
+    hasError,
+    errorType,
+    title,
+  } = useNode((node) => ({
+    url: node.data.props.url,
+    height: node.data.props.height,
+    id: node.id,
+    title: node.data.props.title,
+    hasError: node.data.props.hasError,
+    errorType: node.data.props.errorType,
+  }));
+
+  return (
+    <Box flexWrap="wrap" display="flex" gap="16px" marginBottom="20px">
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeUrlLabel)}{' '}
+              <IconTooltip
+                icon="info3"
+                content={formatMessage(messages.iframeUrlLabelTooltip)}
+              />
+            </span>
+          }
+          placeholder={formatMessage(messages.iframeUrlPlaceholder)}
+          type="text"
+          value={url}
+          onChange={(value) => {
+            const validation = validateUrl(value);
+            setProp((props) => (props.url = value));
+            setProp((props) => (props.errorType = validation[1]));
+            setProp((props) => (props.hasError = !validation[0]));
+            eventEmitter.emit('contentBuilderError', {
+              [id]: !validation[0],
+            });
+          }}
+        />
+        {hasError && (
+          <Error
+            marginTop="4px"
+            text={
+              errorType === 'invalidUrl'
+                ? formatMessage(messages.iframeInvalidUrlErrorMessage)
+                : formatMessage(messages.iframeWhitelistUrlErrorMessage)
+            }
+          />
+        )}
+      </Box>
+      <Box flex="0 0 100%">
+        <Input
+          label={
+            <span>
+              {formatMessage(messages.iframeHeightLabel)}{' '}
```

Comment:
Would personally use `&nbsp;` instead of `{' '}`, but this works too I guess! Whatever you prefer